ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -193,21 +191,44 @@ func (scbuilder *SiteConfigBuilder) getExtraManifest(dataMap map[string]interfac
 				dataMap[filename] = value
 			}
 		} else {
-			manifestFile, err := scbuilder.fHandler.ReadSourceFile(filePath)
+			manifestFile, err := ReadFile(filePath)
+			if err != nil {
+				return dataMap, err
+			}
+			manifestFileStr := string(manifestFile)
+			dataMap[file.Name()] = manifestFileStr
+		}
+	}
+
+	// Adding End User Extra-manifest
+	if clusterSpec.ExtraManifestPath != "" {
+		files, err = GetFiles(clusterSpec.ExtraManifestPath)
+		if err != nil {
+			return dataMap, err
+		}
+		for _, file := range files {
```

Comment:
The logic for handling user files is the same as handling for our standard extra manifests. This should be a common function used for both. This would also enable template handling for user manifests. If that is not desired a flag can be passed to the common function to suppress it.

Reply:
There's no reason to *not* allow templates in the user-provided dir, though, is there?